extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/discovery/WeatherUndergroundDiscoveryService.java

```diff
@@ -0,0 +1,110 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WeatherUndergroundDiscoveryService} creates things based on the configured location.
+ *
+ * @author Laurent Garnier - Initial Contribution
+ */
+public class WeatherUndergroundDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(WeatherUndergroundDiscoveryService.class);
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_WEATHER);
+    private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
+
+    private LocationProvider locationProvider;
+    private ScheduledFuture<?> discoveryJob;
+    private PointType previousLocation;
+
+    /**
+     * Creates a WeatherUndergroundDiscoveryService with enabled autostart.
+     */
+    public WeatherUndergroundDiscoveryService() {
+        super(SUPPORTED_THING_TYPES, DISCOVER_TIMEOUT_SECONDS, true);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting Weather Underground discovery scan");
+        PointType location = locationProvider.getLocation();
+        if (location == null) {
+            logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
+            return;
+        }
+        createResults(location);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (discoveryJob == null) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(() -> {
+                PointType currentLocation = locationProvider.getLocation();
+                if (!Objects.equals(currentLocation, previousLocation)) {
+                    logger.debug("Location has been changed from {} to {}: Creating new discovery results",
+                            previousLocation, currentLocation);
+                    createResults(currentLocation);
+                    previousLocation = currentLocation;
+                }
+            }, 0, LOCATION_CHANGED_CHECK_INTERVAL, TimeUnit.SECONDS);
+            logger.debug("Scheduled Weather Underground location-changed job every {} seconds",
+                    LOCATION_CHANGED_CHECK_INTERVAL);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping Weather Underground device background discovery");
+        if (discoveryJob != null && !discoveryJob.isCancelled()) {
+            if (discoveryJob.cancel(true)) {
+                discoveryJob = null;
+                logger.debug("Stopped Weather Underground device background discovery");
+            }
+        }
+    }
+
+    public void createResults(PointType location) {
+        ThingUID localWeatherThing = new ThingUID(THING_TYPE_WEATHER, LOCAL);
+        Map<String, Object> properties = new HashMap<>(3);
+        properties.put(APIKEY, "ToBeDefined");
```

Comment:
In fact, I thing the discovery service should not set the API key property.
I previously added this property because without it, the thing handler was not started.